### PR TITLE
integration with x509-identity-mgmt

### PR DESCRIPTION
**Pull request by jonaphael**
* **Please check if the PR fulfills these requirements**
  * [ ] Tests for the changes have been added (for bug fixes / features)
  * [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

* **Other information**:

### CODE REVIEW

File: deploy_100k.yaml [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
-- hosts: dojot-k8s[0]
-  any_errors_fatal: true
-  tags: dojot-metrics-prometheus-grafana
-  roles:
-    - role: metrics
+#- hosts: dojot-k8s[0]
+#  any_errors_fatal: true
+#  tags: dojot-metrics-prometheus-grafana
+#  roles:
+#    - role: metrics
```
Comment (eduardogmisiuk) [inline deploy_100k.yaml:10]: This should not be commented.

File: deploy_100k.yaml [hunk 2]
```diff
@@ -63,8 +72,8 @@
     - role: apigw
     - role: auth
     - role: data-broker
-    - role: ejbca-node
+    - role: x509-identity-mgmt
     - role: v2k-bridge
     - role: k2v-bridge
-    - role: latency-pub-test-prometheus
+            # - role: latency-pub-test-prometheus
```
Comment (eduardogmisiuk) [inline deploy_100k.yaml:78]: This can be commented, but correct identation.

File: roles/auth/templates/auth_deployment.yaml.j2 [hunk 3]
```diff
@@ -15,7 +15,8 @@ spec:
         app: dojot
     spec:
       containers:
-      - image: dojot/auth:{{ dojot_auth_version }}
+      #- image: dojot/auth:{{ dojot_auth_version }}
+      - image: thiagodpf/auth:1.0
```
Comment (eduardogmisiuk) [inline roles/auth/templates/auth_deployment.yaml.j2:19]: Leave only the dojot image.

File: roles/k2v-bridge/templates/k2v_bridge_cluster.yaml.j2 [hunk 4]
```diff
@@ -16,9 +16,12 @@ spec:
     spec:
       containers:
       - name: k2v-bridge
-        image: dojot/k2v-bridge:{{ dojot_k2v_version}}
+       # image: dojot/k2v-bridge:{{ dojot_k2v_version}}
+        image: jonaphael/k2v-x509-identity-mgmt:development
```
Comment (eduardogmisiuk) [inline roles/k2v-bridge/templates/k2v_bridge_cluster.yaml.j2:20]: Leave only the dojot image.

File: roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2 [hunk 5]
```diff
@@ -25,10 +25,11 @@ spec:
                 - verne-kafka
       containers:
       - name: v2k-bridge
-        image: dojot/v2k-bridge:{{ dojot_v2k_version}}
+        #image: dojot/v2k-bridge:{{ dojot_v2k_version}}
+        image: jonaphael/v2k-x509-identity-mgmt:development
```
Comment (eduardogmisiuk) [inline roles/v2k-bridge/templates/v2k_bridge_cluster.yaml.j2:29]: Leave only the dojot image.

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 [hunk 6]
```diff
@@ -6,7 +6,8 @@ metadata:
   name: k8s
   namespace: {{ dojot_namespace }}
 spec:
-  baseImage: dojot/vernemq-dojot
+  # baseImage: dojot/vernemq-dojot
+  baseImage: jonaphael/verne-x509-identity-mgmt
```
Comment (eduardogmisiuk) [inline roles/vernemq/templates/z_vernemq_cluster.yaml.j2:10]: Leave only the dojot image.

File: roles/vernemq/templates/z_vernemq_cluster.yaml.j2 [hunk 7]
```diff
@@ -57,5 +58,6 @@ spec:
     plugins: []
   serviceAccountName: vernemq-k8s
   size: {{ dojot_vernemq_mqtt_replicas_to_scale }}
-  tag: {{ dojot_verne_version }}
+  #tag: {{ dojot_verne_version }}
+  tag: development
```
Comment (eduardogmisiuk) [inline roles/vernemq/templates/z_vernemq_cluster.yaml.j2:62]: Leave only the dojot image tag.

File: roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2 [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            values:
+            - verne-kafka
+  selector:
+    matchLabels:
+      app: x509-identity-mgmt
+  replicas: 1
+  template: #templates for the pods
+    metadata:
+      labels:
+        app: x509-identity-mgmt
```
Comment (eduardogmisiuk) [inline roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2:24]: In the Ansible I'm running for the 100K epic, I changed the tags to something like this:
```yaml
app: dojot
name: x509-identity-mgmt
```

This way, we can filter all logs for Dojot, having a similar effect as docker-compose logs. This behaviour is useful for some tests, as @betefaber said to me.
Don't forget to change the selector in the service file.

File: roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2 [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    spec:
+      hostname: "x509-identity-mgmt"
+      #subdomain: "dojot.iot"
+      containers:
+      - name: x509-identity-mgmt
+        image: thiagodpf/x509-identity-mgmt:1.2
+        #image: jonaphael/x509-identity-mgmt:development
+        #image: dojot/x509-identity-mgmt:{{ dojot_version  }}
```
Comment (eduardogmisiuk) [inline roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2:32]: Leave only the dojot image.

File: roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2 [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        - name: MONGO_URI
+          value: "mongodb://mongodb:27017/x509-identity-mgmt"
+      volumeMounts:
+      - mountPath: /mnt/persistent
+        name: x509-idenity-data
+      restartPolicy: Always
+      volumes:
+      - name: x509-idenity-data
+        persistentVolumeClaim:
+          claimName: x509-identity-pvc
```
Comment (eduardogmisiuk) [inline roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2:53]: If you've already made the PV and PVC files, why do you create the volumes here too? The way we are doing in the 100K Ansible used in tests is to separate in files, I think we can remove this and only use the files.

Comment (jonaphael) [reply roles/x509-identity-mgmt/templates/x509_identity_mgmt_cluster.yaml.j2:53]: This volume is just for mapping the deployment to the persistent volume claim.

**Pull request closed without merging**